cmd/backup/restore.go

```diff
@@ -81,3 +93,64 @@ func getTargetTime() (time.Time, error) {
 func writeTargetFile(backupTargetFilePath string) error {
 	return os.WriteFile(targetFilePath, []byte(backupTargetFilePath), 0777)
 }
+
+func uncompressFile(f string) error {
+	tmpf := f + ".tmp"
+	err := os.Rename(f,tmpf)
+	if err != nil {
+		panic(err)
+	}
+	compressedFile, err := os.Open(tmpf)
+	if err != nil {
+		return (err)
+	}
+	defer compressedFile.Close()
+
+	originalFile, err := os.Create(f)
+	if err != nil {
+		return (err)
+	}
+	defer originalFile.Close()
+
+	bReader := bufio.NewReader(compressedFile)
+	testBytes, err := bReader.Peek(2)
+	if err != nil {
+		return (err)
+	}
+	if testBytes[0] == 31 && testBytes[1] == 139 {
```

Comment:
Although I really appreciated the effort here, I think this will be difficult to maintain and scale to have more compression algorithms. 

I suggest that, instead of checking the contents of the file, we could add the compression algorithm to the backup filename: `backup.2024-08-26T12:24:34Z.{gzip,zlib}.sq`l. We are handling this here:
- https://github.com/mariadb-operator/mariadb-operator/blob/6171b061f944e10e29197fbdc7e4d1144b08352d/pkg/backup/backup.go#L69
- https://github.com/mariadb-operator/mariadb-operator/blob/6171b061f944e10e29197fbdc7e4d1144b08352d/pkg/command/backup.go#L226

This way, the name of the file has all the metadata explicitly to operate with it, it is human readable and there is no need to check its contents.

It is important to note that, this should be backwards compatible, the process should allow file names without compression.